### DISCUSSION TITLE
Is it possible to dump (at least part of) the learned lemmas?

### DISCUSSION THREAD

**Discussion by aur3l14no** [cvc5/cvc5#8778] [Q&A] [answered]
I'm wondering if it is possible to dump part of the learned lemmas for debugging purposes.

I tried tracing "prop::lemmas" and "theory::lemma", which give me some lemmas involving skolem functions. Is it correct? Also, I don't know how to dump the declarations of these skolem functions (and their types).

Thanks!

**Comment by ajreynol** [comment, marked as answer]
If you have a trace-enabled build, `-t im` is the authoritative trace on all lemmas, conflicts and facts inferred by theory solvers (a "fact" is an inference step that is internal to a theory solver and not communicated to the SAT solver).  The output format of this trace will give identifiers and the inferred formula in question, where the possible identifiers are listed here: https://github.com/cvc5/cvc5/blob/main/src/theory/inference_id.h#L41

Also, we have a new feature `-o learned-literals` that outputs when a unit clause is learned by the SAT solver, i.e. the literal holds globally.  For details on this see:  https://github.com/cvc5/cvc5/blob/main/src/api/cpp/cvc5_types.h#L136
